haystack/document_store/faiss.py

```diff
@@ -79,13 +79,14 @@ def __init__(
                              Can be helpful to disable in production deployments to keep the logs clean.
         """
         self.vector_dim = vector_dim
-
+        self.faiss_index_factory_str = faiss_index_factory_str
+        self.faiss_indexes: Dict[str, faiss.swigfaiss.Index] = {}
         if faiss_index:
-            self.faiss_index = faiss_index
+            self.faiss_indexes[index] = faiss_index
         else:
-            self.faiss_index = self._create_new_index(vector_dim=self.vector_dim, index_factory=faiss_index_factory_str, **kwargs)
-            if "ivf" in faiss_index_factory_str.lower():  # enable reconstruction of vectors for inverted index
-                self.faiss_index.set_direct_map_type(faiss.DirectMap.Hashtable)
+            self.faiss_indexes[index] = self._create_new_index(
```

Comment:
This is what my original design to have faiss_index per index, good we are moving to that direction. :)